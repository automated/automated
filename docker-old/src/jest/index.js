"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.runner = void 0;
const change_case_1 = require("change-case");
const snapshot_diff_1 = __importDefault(require("snapshot-diff"));
const jest_image_snapshot_1 = require("jest-image-snapshot");
const derive_describe_name_1 = __importDefault(require("../utils/derive-describe-name"));
const derive_use_cases_1 = __importDefault(require("../utils/derive-use-cases"));
const shared = require('../storybook/shared');
const puppeteer_1 = __importDefault(require("puppeteer"));
const react_test_renderer_1 = __importDefault(require("react-test-renderer"));
const react_1 = __importDefault(require("react"));
const screenshot_configs_1 = __importDefault(require("./screenshot-configs"));
expect.extend({
    toMatchImageSnapshot: jest_image_snapshot_1.toMatchImageSnapshot,
});
const runner = async ({ dirname, Component, useCases: useCasesProp, }) => {
    const describeName = (0, derive_describe_name_1.default)({ dirname });
    const useCases = (0, derive_use_cases_1.default)({ useCases: useCasesProp });
    let isStorybookRunning = process.env.AUTOMATED_STORYBOOK_IS_RUNNING;
    let storybookTestFn = isStorybookRunning ? test : test.skip;
    let browser;
    beforeAll(async () => {
        if (isStorybookRunning) {
            browser = await puppeteer_1.default.launch({
                // headless: false,
                args: ['--no-sandbox'],
            });
        }
        else {
            if (process.env.AUTOMATED_JEST_VISUAL_REGRESSION_REQUIRED) {
                throw new Error('`AUTOMATED_JEST_VISUAL_REGRESSION_REQUIRED` is true, ' +
                    'but visual regression is disabled');
            }
        }
    });
    describe(describeName, () => {
        useCases.forEach(async (item, key) => {
            const { name, props } = item;
            afterAll(async () => {
                if (browser)
                    await browser.close();
            });
            test(`snapshot-${name}`, async () => {
                const render = react_test_renderer_1.default.create(react_1.default.createElement(Component, { ...props }));
                if (!key) {
                    expect(react_1.default.createElement(Component, { ...props })).toMatchSnapshot();
                }
                else {
                    expect((0, snapshot_diff_1.default)(react_1.default.createElement(Component, { ...useCases[0].props }), react_1.default.createElement(Component, { ...props }))).toMatchSnapshot();
                }
            });
            Object.entries(screenshot_configs_1.default).forEach(([configName, { viewport, matchImageSnapshotOptions, screenshotOptions },]) => {
                storybookTestFn(`storybook@${describeName}@${name}@${configName}`, async () => {
                    if (browser) {
                        const page = await browser.newPage();
                        await page.evaluateOnNewDocument(() => {
                            const style = document.createElement('style');
                            style.innerHTML = '.body { caret-color: transparent }';
                            document.getElementsByTagName('head')[0].appendChild(style);
                        });
                        const storybookId = `${(0, change_case_1.paramCase)(describeName)}--${name}`;
                        const url = `${shared.getStorybookUrl()}/iframe.html?id=${storybookId}&args=&viewMode=story`;
                        await page.goto(url);
                        await page.setViewport(viewport);
                        const image = await page.screenshot(screenshotOptions);
                        expect(image).toMatchImageSnapshot(matchImageSnapshotOptions);
                        await page.close();
                    }
                });
            });
        });
    });
};
exports.runner = runner;
//# sourceMappingURL=index.js.map