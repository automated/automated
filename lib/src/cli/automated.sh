COMMAND="$(pwd)/node_modules/@automated/automated/lib/dist/cli/runner.js $@"
echo x2
echo $AUTOMATED_JEST_VISUAL_REGRESSION_REQUIRED
node $COMMAND
