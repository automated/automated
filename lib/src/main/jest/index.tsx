import type ChangeCase from 'change-case';
import type JestImageSnapshot from 'jest-image-snapshot';
import type Puppeteer from 'puppeteer/lib/types';
import React from 'react';
import type TestRenderer from 'react-test-renderer';
import type SnapshotDiff from 'snapshot-diff';

import { deriveModule } from '../shared';
import { getStorybookUrl } from '../storybook/shared';
import { UseCases } from '../types';
import deriveDescribeName from '../utils/derive-describe-name';
import deriveUseCases from '../utils/derive-use-cases';
import screenshotConfigs from './screenshot-configs';

// eslint-disable-next-line import/prefer-default-export
export const runner = async ({
  dirname,
  Component,
  useCases: useCasesProp,
}: {
  dirname: string;
  Component: React.ElementType;
  useCases?: UseCases;
}) => {
  const { paramCase }: typeof ChangeCase = await deriveModule('change-case');
  const snapshotDiff: typeof SnapshotDiff = await deriveModule('snapshot-diff');
  const { toMatchImageSnapshot }: typeof JestImageSnapshot = await deriveModule(
    'jest-image-snapshot',
  );
  const puppeteer: typeof Puppeteer = await deriveModule('puppeteer');
  const testRenderer: typeof TestRenderer = await deriveModule(
    'react-test-renderer',
  );

  expect.extend({
    toMatchImageSnapshot,
  });

  const describeName = deriveDescribeName({ dirname });
  const useCases = deriveUseCases({ useCases: useCasesProp });

  const isStorybookRunning = process.env.AUTOMATED_STORYBOOK_IS_RUNNING;
  const storybookTestFn = isStorybookRunning ? test : test.skip;
  let browser: Puppeteer.Browser;

  beforeAll(async () => {
    if (isStorybookRunning) {
      browser = await puppeteer.launch({
        // headless: false,
        args: ['--no-sandbox'],
      });
    } else if (process.env.AUTOMATED_JEST_VISUAL_REGRESSION_REQUIRED) {
      throw new Error(
        '`AUTOMATED_JEST_VISUAL_REGRESSION_REQUIRED` is true, ' +
          'but visual regression is disabled',
      );
    }
  });

  describe(describeName, () => {
    useCases.forEach(async (item, key) => {
      const { name, props } = item;

      afterAll(async () => {
        if (browser) await browser.close();
      });

      test(`snapshot-${name}`, async () => {
        const render = testRenderer.create(<Component {...props} />).toJSON();

        if (!key) {
          expect(render).toMatchSnapshot();
        } else {
          expect(
            snapshotDiff(<Component {...useCases[0].props} />, render),
          ).toMatchSnapshot();
        }
      });

      Object.entries(screenshotConfigs).forEach(
        ([
          configName,
          { viewport, matchImageSnapshotOptions, screenshotOptions },
        ]) => {
          storybookTestFn(
            `storybook@${describeName}@${name}@${configName}`,
            async () => {
              if (browser) {
                const page = await browser.newPage();

                await page.evaluateOnNewDocument(() => {
                  const style = document.createElement('style');
                  style.innerHTML = '.body { caret-color: transparent }';
                  document.getElementsByTagName('head')[0].appendChild(style);
                });

                const storybookId = `${paramCase(describeName)}--${name}`;

                const url = `${getStorybookUrl()}/iframe.html?id=${storybookId}&args=&viewMode=story`;
                await page.goto(url);
                await page.setViewport(viewport);

                const image = await page.screenshot(screenshotOptions);

                expect(image).toMatchImageSnapshot(matchImageSnapshotOptions);

                await page.close();
              }
            },
          );
        },
      );
    });
  });
};
