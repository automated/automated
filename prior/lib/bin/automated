#!/bin/bash

export NODE_ENV="automated"

### Get cli arguments
# maintain arg quotes - https://stackoverflow.com/a/24179878/537998
function requote() {
  local res=""
  for x in "${@}" ; do
      # try to figure out if quoting was required for the $x:
      grep -q "[[:space:]]" <<< "$x" && res="${res} '${x}'" || res="${res} ${x}"
  done
  # remove first space and print:
  sed -e 's/^ //' <<< "${res}"
}
cliArgs=$(requote "${@}")
###

pwd=`pwd`;
pluginDirSearch='/node_modules/@automated/[plugin-]*'

declare -a lifecycles=(
  collection-will-start
  collection-did-start
  collection-running
  test-will-start
  test-did-start
  test-running
  test-will-end
  collection-will-end
)

for lifecycle in "${lifecycles[@]}"
do
  for pluginDir in `ls -d $pwd$pluginDirSearch`
  do
    # look for something like `jest[`
    pluginWithBraket=$(echo $pluginDir | pcregrep -o1 'plugin-(.*)')\\[
    script=$pluginDir/$lifecycle
    if [[ -e $script ]]; then
      if [[ $cliArgs == *$pluginWithBraket* ]]; then
        # run with args
        pluginArgs=$(echo $cliArgs | pcregrep -o1 "$pluginWithBraket(.*)\]")
        cmd=$(echo $script $pluginArgs)
        # echo $cmd
        $cmd
      else
        # run without args
        cmd=$(echo $script)
        # echo $cmd
        $cmd
      fi
    fi
  done
done
