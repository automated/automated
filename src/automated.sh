COMMAND="$(pwd)/node_modules/@automated/automated/dist/cli/runner.js $@"
node $COMMAND
