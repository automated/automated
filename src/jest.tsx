import { Browser } from 'puppeteer/lib/types';
// import { render } from '@testing-library/react';
import { toMatchDiffSnapshot } from 'snapshot-diff';
import { toMatchImageSnapshot } from 'jest-image-snapshot';
import { UseCases } from './types';
import puppeteer from 'puppeteer';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import { paramCase } from 'change-case';

expect.extend({ toMatchImageSnapshot, toMatchDiffSnapshot });

const base = ({
  useCases,
  Component,
  describeName,
}: {
  useCases: UseCases;
  Component: React.ElementType;
  describeName: string;
}) => {
  let browser: Browser;

  beforeAll(async () => {
    browser = await puppeteer.launch({
      headless: false,
    });
  });

  afterAll(async () => {
    await browser.close();
  });

  describe(describeName, () => {
    Object.entries(useCases).forEach(([key, value]) => {
      const { props } = value;

      // test(`snapshot-${key}`, async () => {
      //   const render = TestRenderer.create(<Component {...props} />);
      //   const renderToJson = render.toJSON();
      //   if (key === 'default') renderToJson;
      //   expect(renderToJson).toMatchSnapshot();
      // });

      test(`image-${key}`, async () => {
        const page = await browser.newPage();

        const url = `http://localhost:3144/iframe.html?id=${paramCase(
          describeName,
        )}--${key}&args=&viewMode=story`;
        console.log(url);
        await page.goto(url);

        // caret-color: transparent;

        // await page.screenshot({ path: `example-${key}.png` });
        const image = await page.screenshot();

        await page.close();

        expect(image).toMatchImageSnapshot();
      });
    });
  });
};

export default base;
