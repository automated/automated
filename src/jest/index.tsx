import { Browser } from 'puppeteer/lib/types';
import { paramCase } from 'change-case';
import { toMatchDiffSnapshot } from 'snapshot-diff';
import { toMatchImageSnapshot } from 'jest-image-snapshot';
import { UseCases } from '../types';
import deriveDescribeName from '../utils/derive-describe-name';
import deriveUseCases from '../utils/derive-use-cases';
import shared from '../shared';
import puppeteer from 'puppeteer';
import TestRenderer from 'react-test-renderer';
import React from 'react';
import screenshotConfig from './screenshot-config';

expect.extend({ toMatchImageSnapshot, toMatchDiffSnapshot });

export const runner = async ({
  dirname,
  Component,
  useCases: useCasesProp,
}: {
  dirname: string;
  Component: React.ElementType;
  useCases?: UseCases;
}) => {
  const describeName = deriveDescribeName({ dirname });
  const useCases = deriveUseCases({ useCases: useCasesProp });

  let isStorybookRunning = process.env.STORYBOOK_IS_RUNNING;
  let storybookTestFn = isStorybookRunning ? test : test.skip;
  let browser: Browser;

  beforeAll(async () => {
    if (isStorybookRunning) {
      browser = await puppeteer.launch({
        // headless: false,
      });
    }
  });

  describe(describeName, () => {
    useCases.forEach(async (item, key) => {
      const { name, props } = item;

      afterAll(async () => {
        if (browser) await browser.close();
      });

      test(`snapshot-${name}`, async () => {
        const render = TestRenderer.create(<Component {...props} />);
        const renderToJson = render.toJSON();
        if (!key) renderToJson;
        expect(renderToJson).toMatchSnapshot();
      });

      console.log(screenshotConfig);

      storybookTestFn(`storybook-${name}`, async () => {
        if (browser) {
          const page = await browser.newPage();

          await page.evaluateOnNewDocument(() => {
            const style = document.createElement('style');
            style.innerHTML = '.body { caret-color: transparent }';
            document.getElementsByTagName('head')[0].appendChild(style);
          });

          const id = `${paramCase(describeName)}--${name}`;

          const url = `${shared.storybookUrl}/iframe.html?id=${id}&args=&viewMode=story`;
          await page.goto(url);
          await page.setViewport({
            deviceScaleFactor: 2,
            height: 768,
            width: 1080,
          });

          const image = await page.screenshot({
            omitBackground: true,
          });

          expect(image).toMatchImageSnapshot({
            failureThreshold: 0.01,
            failureThresholdType: 'percent',
          });

          await page.close();
        }
      });
    });
  });
};
