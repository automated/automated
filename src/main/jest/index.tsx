import React from 'react';

import type Puppeteer from 'puppeteer/lib/types';
import type SnapshotDiff from 'snapshot-diff';
import type ChangeCase from 'change-case';
import type JestImageSnapshot from 'jest-image-snapshot';

import deriveDescribeName from '../utils/derive-describe-name';
import { UseCases } from '../types';
import screenshotConfigs from './screenshot-configs';
import deriveUseCases from '../utils/derive-use-cases';
import { getStorybookUrl } from '../storybook/shared';
import { deriveModule } from '../shared';

export const runner = async ({
  dirname,
  Component,
  useCases: useCasesProp,
}: {
  dirname: string;
  Component: React.ElementType;
  useCases?: UseCases;
}) => {
  const { paramCase }: typeof ChangeCase = deriveModule('change-case');
  const snapshotDiff: typeof SnapshotDiff =
    deriveModule('snapshot-diff').default;
  const { toMatchImageSnapshot }: typeof JestImageSnapshot = deriveModule(
    'jest-image-snapshot',
  );
  const puppeteer: typeof Puppeteer = deriveModule('puppeteer');
  // import TestRenderer from 'react-test-renderer';

  expect.extend({
    toMatchImageSnapshot,
  });

  const describeName = deriveDescribeName({ dirname });
  const useCases = deriveUseCases({ useCases: useCasesProp });

  let isStorybookRunning = process.env.AUTOMATED_STORYBOOK_IS_RUNNING;
  let storybookTestFn = isStorybookRunning ? test : test.skip;
  let browser: Puppeteer.Browser;

  beforeAll(async () => {
    if (isStorybookRunning) {
      browser = await puppeteer.launch({
        // headless: false,
        args: ['--no-sandbox'],
      });
    } else {
      if (process.env.AUTOMATED_JEST_VISUAL_REGRESSION_REQUIRED) {
        throw new Error(
          '`AUTOMATED_JEST_VISUAL_REGRESSION_REQUIRED` is true, ' +
            'but visual regression is disabled',
        );
      }
    }
  });

  describe(describeName, () => {
    useCases.forEach(async (item, key) => {
      const { name, props } = item;

      afterAll(async () => {
        if (browser) await browser.close();
      });

      test(`snapshot-${name}`, async () => {
        // const render = TestRenderer.create(<Component {...props} />);

        if (!key) {
          expect(<Component {...props} />).toMatchSnapshot();
        } else {
          expect(
            snapshotDiff(
              <Component {...useCases[0].props} />,
              <Component {...props} />,
            ),
          ).toMatchSnapshot();
        }
      });

      Object.entries(screenshotConfigs).forEach(
        ([
          configName,
          { viewport, matchImageSnapshotOptions, screenshotOptions },
        ]) => {
          storybookTestFn(
            `storybook@${describeName}@${name}@${configName}`,
            async () => {
              if (browser) {
                const page = await browser.newPage();

                await page.evaluateOnNewDocument(() => {
                  const style = document.createElement('style');
                  style.innerHTML = '.body { caret-color: transparent }';
                  document.getElementsByTagName('head')[0].appendChild(style);
                });

                const storybookId = `${paramCase(describeName)}--${name}`;

                const url = `${getStorybookUrl()}/iframe.html?id=${storybookId}&args=&viewMode=story`;
                await page.goto(url);
                await page.setViewport(viewport);

                const image = await page.screenshot(screenshotOptions);

                expect(image).toMatchImageSnapshot(matchImageSnapshotOptions);

                await page.close();
              }
            },
          );
        },
      );
    });
  });
};
