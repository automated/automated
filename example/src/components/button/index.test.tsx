import { runner } from '@automated/automated/out/jest/index';
import automated from './automated';

runner(automated);
