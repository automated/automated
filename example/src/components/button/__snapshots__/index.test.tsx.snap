// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/Users/kirkstrobeck/sync/homedir/git/automated/automated/example/src/components/button/index.test.tsx default 1`] = `
.emotion-0 {
  padding: 20px 30px;
  border-radius: 10px;
  border: none;
  background: blue;
  color: white;
  font-weight: 600;
  font-size: 20px;
}

<button
  className="emotion-0"
  onClick={[Function]}
>
  Get started
</button>
`;

exports[`/Users/kirkstrobeck/sync/homedir/git/automated/automated/example/src/components/button/index.test.tsx setBackground 1`] = `
.emotion-0 {
  padding: 20px 30px;
  border-radius: 10px;
  border: none;
  background: green;
  color: white;
  font-weight: 600;
  font-size: 20px;
}

<button
  className="emotion-0"
  onClick={[Function]}
>
  Get started
</button>
`;

exports[`/Users/kirkstrobeck/sync/homedir/git/automated/automated/example/src/components/button/index.test.tsx setText 1`] = `
.emotion-0 {
  padding: 20px 30px;
  border-radius: 10px;
  border: none;
  background: blue;
  color: white;
  font-weight: 600;
  font-size: 20px;
}

<button
  className="emotion-0"
  onClick={[Function]}
>
  Hello automated
</button>
`;
